keps/sig-node/3953-node-resource-hot-plug/README.md

```diff
@@ -0,0 +1,862 @@
+
+Yes. The feature can be disabled by restarting kubelet with the feature-gate off.
+Once disabled any hot plug of resources won't reflect at the cluster level.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+To reenanble the feature, need to turn on the feature-gate and restart the kubelet,
+with feature reenabled, the node resources can be hot plugged in again. Cluster will be automatically updated
+with the new resource information. If there are any pending pods due to lack of resources they will turn into
+running state.
+
+###### Are there any tests for feature enablement/disablement?
+
+Yes, the tests will be added along with alpha implementation.
+* Validate the hot plug of resource to machine is updated at the node resource level.
+* Validate the hot plug of resource made the pending pods to transition into running state.
+* Validate the resource managers are update with the latest machine information after hot plug of resources.
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+Rollout may fail if the resource managers are not re-synced properly due to programmatic errors.
+In case of rollout failures, running workloads are not affected, If the pods are on pending state they remain
+in the pending state only.
+Rollback failure should not affect running workloads.
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+If there is significant increase in `node_resize_resync_errors_total` metric means the feature is not working as expected.
+In case of pending pods and hot plug of resource but still there is no change `scheduler_pending_pods` metric
+means the feature is not working as expected.
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+It will be tested manually as a part of implementation and there will also be automated tests to cover the scenarios.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+No
```

Comment:
Would we add a flag to control the poll duration about kubelet reading new machine info from cAdvisor

Reply:
Flag, possibly not. Did you mean "field within the kubelet configuration file?"

(using command line arguments to kubelet is mostly deprecated in favor of using configuration fields)

Reply:
As suggested in [comment](https://github.com/kubernetes/enhancements/pull/3955#discussion_r1955119053) , we have decided to move away from poll mechanism to level triggered mechanism, Where we continuously watch for resize changes from cAdvisor and handle them accordingly.